src/function.ts

```diff
@@ -42,34 +40,32 @@ export type AnyLambda = Function<AnyFunction>;
  *
  * Make sure to implement your Lambda Function entry-points accordingly.
  */
-export class Function<F extends AnyFunction> {
+export class Function<P = undefined, O = void> {
   readonly kind: "Function" = "Function";
 
-  // @ts-ignore - this makes `F` easily available at compile time
-  readonly __functionBrand: F;
-
-  constructor(
-    readonly resource: aws_lambda.IFunction,
-    /**
-     * Names of the arguments in order.
-     * If this is omitted, then it will be injected by a TS transform.
-     */
-    readonly args: string[] = []
-  ) {
+  constructor(readonly resource: aws_lambda.IFunction) {
     return Object.assign(lambda, this);
 
     function lambda(call: CallExpr, vtl: VTL): string {
-      const payload = vtl.var(`{}`);
-      for (const [argName, argVal] of Object.entries(call.args)) {
-        vtl.qr(`${payload}.put('${argName}', ${vtl.eval(argVal)})`);
-      }
+      // first argument is the payload
+      const [payloadKey] = Object.keys(call.args);
```

Comment:
Could it be undefined?

Reply:
no, because its now the definition of the Function, so I was always getting `args`.

Reply:
What about when P is undefined? Didn't you change the signature to have no args in that case?

Reply:
it still returns `args` as a property name (the tests seems to work).

Reply:
Upon more testing, the args map is always returning `args` mapped to either `nullliteral` or a prop accessor, base on 0 or one arguments. Given two arguments it still returns only one.

Putting function brand back and set to the computed `ConditionalFuntion` has returned a more logical result. Will refacto r :+1:

Reply:
Yes function brand is required to give the type of the resolved function to the compiler.